### ISSUE TITLE
[INSTALL]: Try Python 3.11.2+

### ISSUE THREAD

**Issue by DavidHuber-NOAA** [JCSDA/spack-stack#908]
### Package name

Python

### Package version/tag

3.11.2+

### Build options

Standard options

### Installation timeframe

With spack-stack 1.6.0.

### Other information

This is needed for a bug fix that was never incorporated into 3.10.x that is breaking metplus/3.1.1.

### WCOSS2

- [ ] Check this box if and only if your package should be installed on WCOSS2 Cactus and Dogwood (all spack-stack packages will be installed on Acorn). If not, you may disregard the rest of the items below and submit this request.

### WCOSS2: Technical & security review list

- [ ] The code is mature, stable, and production ready
- [ ] The code is does not and cannot use the internet, and does not contain URLs (http, https, ftp, etc.) except in comments
- [ ] The package does not contain prebuilt binary files that have not been approved by NCO security review
- [ ] The code has no publicly disclosed cybersecurity vulnerabilities and exposures (search https://cve.mitre.org/cve/)
- [ ] The code is not prohibited by DHS, DOC, NOAA, or NWS
- [ ] The code comes from a trusted source. Trusted sources include other NWS, NOAA, or DOC, agencies, or other Federal agencies that operate at a FISMA high or equivalent level. Additionally, trusted sources could be third-party agencies through which there is an existing SLA on file (such as RedHat).
- [ ] The code is actively maintained and supported (it continues to get updates, patches, etc.)
- [ ] The code is not maintained by a private entity operating in a foreign country (if it is, make a note below)
- [ ] There is sufficient documentation to support maintenance
- [ ] There are no known security vulnerabilities or weaknesses
- [ ] Installing and running the code does not require privileged processes/users
- [ ] There are no software dependencies that are unapproved or have security concerns (if there are, make a note below)
- [ ] There are no concerns related to SA, SI, and SC NIST control families

**Comment by climbfuji** [comment]
@DavidHuber-NOAA Unfortunately we can't bump Python to 3.11.2 for the unified environment. Many Python packages, especially for the JEDI EWOK environment, don't build with Python 3.11 (or at least the package maintainers explicitly wrote that in the package recipes), i.e. the `concretize` step fails right away.

What I can try is to change the Python version for the `gsi-addon` environment only. Would that be useful for your transition to the new EVS system (sorry if I got that acronym wrong)?

**Comment by DavidHuber-NOAA** [comment]
@climbfuji Yes, that could work for `verif-global`.  Thanks!

**Comment by climbfuji** [comment]
So far I haven't been successful. I've updated many packge version in the environment, because many of our default dependencies (versions) don't build with 3.11.2. I'll try a bit more, but then I'll give up.

**Comment by DavidHuber-NOAA** [comment]
Alright, thank you, @climbfuji.  I'm willing to accept a null result, but appreciate the effort!

**Comment by climbfuji** [comment]
I am not getting anywhere with this on my macOS. It's an older version of the OS (Monterey) and the clang compilers (13.1.6), but I am not sure if that's the problem. The issue is that `py-numpy` doesn't work correctly, which then breaks the install of many packages that rely on `py-numpy` to build:
```
> python3
Python 3.11.2 (main, Dec 15 2023, 08:07:15) [Clang 13.1.6 (clang-1316.0.21.2.3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/heinzell/work/spack-stack/spack-stack-add-esmf860/envs/gsi-addon-env/install/apple-clang/13.1.6/py-numpy-1.26.1-zjmtp4y/lib/python3.11/site-packages/numpy/__init__.py", line 404, in <module>
    raise RuntimeError(msg)
RuntimeError: Polyfit sanity test emitted a warning, most likely due to using a buggy Accelerate backend.
If you compiled yourself, more information is available at:
https://numpy.org/doc/stable/user/building.html#accelerated-blas-lapack-libraries
Otherwise report this to the vendor that provided NumPy.
RuntimeWarning: invalid value encountered in divide
```
After checking with Prof. Google, I verified that the blas/lapack backends are all passed to the numpy build correctly (I think). I created an issue in the spack github space, https://github.com/spack/spack/issues/41742.  I am also going to test this on Linux as well as on my other mac laptop (Sonoma with clang 15.0.0) before I throw in the towel.

**Comment by climbfuji** [comment]
Unfortunately this won't be done in time for spack-stack-1.6.0 (but shortly thereafter).

**Comment by climbfuji** [comment]
Closed via #920
